src/server/IoDispatchers.cpp

```diff
@@ -178,27 +178,15 @@ static bool _shouldAttemptHandoff(const Globals& globals,
 
 #else
 
-    // Service desktops and non-interactive sessions should not
-    // try to hand off -- they probably don't have any terminals
-    // installed, and we don't want to risk breaking a service if
-    // they *do*.
-    if (!_isInteractiveUserSession())
-    {
-        return false;
-    }
-
-    // This console was started with a command line argument to
-    // specifically block handoff to another console. We presume
-    // this was for good reason (compatibility) and give up here.
-    if (globals.launchArgs.GetForceNoHandoff())
+    // If we do not have a registered handoff, do not attempt.
+    if (!globals.handoffConsoleClsid)

```

Comment:
I noticed a slight "bug" (I don't think it can occur tbh) in my first commit and added a `Globals.handoffConsoleClsid.reset();` below in my last commit.
While doing so I noticed that the order of if checks in `_shouldAttemptHandoff` is inoptimal: It first checked `_isInteractiveUserSession` instead of checking cheap boolean flags first, so I changed this code. Let me know if I should revert it for a more compact PR.